R/commonMachineLearningRegression.R

```diff
@@ -108,6 +108,17 @@
     observations.amount = "< 2",
     exitAnalysisIfErrors = TRUE
   )
+  if (options[["target"]] != "" && length(options[["predictors"]]) > 0) {
+    predictorData <- dataset[, options[["predictors"]], drop = FALSE]
+    if (sum(apply(predictorData, MARGIN = 2, FUN = function(x) is.character(x) || is.factor(x))) > 0) {
```

Comment:
since `predictorData` is a data.frame, you should probably do:
```suggestion
    if (any(sapply(predictorData, FUN = function(x) is.character(x) || is.factor(x))))) {
```
since `apply` does `as.matrix` on the `predictorData` behind the scenes, which is a bit redundant.

I'd prefer `vapply(predictorData, FUN = function(x) is.character(x) || is.factor(x), FUN.VALUE = logical(1L))` but that's up to you.

The same `sapply` rather than `vapply` goes for the check with
```r
factorVarsWithUniqueLevels <- which(apply(predictorData[, sapply(predictorData, function(x) is.character(x) || is.factor(x)), drop = FALSE], MARGIN = 2, FUN = function(x) length(x) == length(unique(x))))
```